### DISCUSSION TITLE
Distributed File System

### DISCUSSION THREAD

**Discussion by pdgetrf** [CentaurusInfra/fornax#80] [Ideas]
General discussion about distributed file system e.g. IPFS

**Comment by pdgetrf** [comment]
From @deepak-vij 

![image](https://user-images.githubusercontent.com/252020/153522412-66616273-c2cc-49cf-84a2-b0a9e853713c.png)

**Comment by deepak-vij** [comment]
As we all know that IPFS does use an information-centric like approach to storage and retrieval through self-certifying names for content. IPFS works as an application-layer level overlay. It can work on top of different underlying networking protocols, such as TCP, UDP or WebRTC operating on top of an endpoint-centric IP.

In contrast, Information-Centric Networking (ICN) solutions like NDN and CCN can reduce network load by natively delivering content from cached copies of the downloaded content at the end users. Essentially, IPFS can be really enhanced by working in conjunction with ICN based approach. That way, it is also not restricted by issues such as application-level resolution mechanism and unicast communication etc.

I am quite sure this is exactly what “Extensible Internet” folks are doing at UC Berkeley. In any case, we will find out more when we all meet with Prof. Shenker towards end of this month.
